packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -751,10 +760,13 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     }
 
     if (widget.enableSearch) {
-      if (widget.searchCallback != null) {
-        currentHighlight = widget.searchCallback!(filteredEntries, _localTextEditingController!.text);
-      } else {
-        currentHighlight = search(filteredEntries, _localTextEditingController!);
+      final bool shouldUpdateCurrentHighlight = _shouldUpdateCurrentHighlight(filteredEntries, _localTextEditingController!);
```

Comment:
We shouldn't use `shouldUpdateCurrentHighlight` when a `searchCallback` is provided, since we don't know its match condition.